foyer/general_forcefield.py

```diff
@@ -0,0 +1,502 @@
+        Creat Angles, Dihedrals, Impropers and assing corresponding
+        AngleType, DihedralTypes, and ImproperTypes.
+
+        Parameters
+        ----------
+        top : gmso.Topology
+            gmso.Topology object that needed to be parametrized
+        typemap : dict
+            typemap generated by the atomtyper
+        references_file : str, optional, default=None
+            Name of the file where force field references will be written
+            (in Bibtex format).
+        assert_bond_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system bonds.
+        assert_angle_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system angles.
+        assert_dihedral_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system dihedrals.
+        assert_improper_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system impropers.
+        combining_rule : str, optional, default='geometric'
+            The combining rule of the system, stored as an attribute of the
+            GMSO topology. Available options: 'geometric', 'lorentz'.
+        debug : bool, optional, default=False
+            If True, Foyer will print debug-level information about notable or
+            potential problematic details it encounters.
+
+        Returns
+        -------
+        top : gmso.Topology
+            A parametrized Topology.
+        """
+        # Generate missing connection objects (angles, dihedrals, and impropers)
+        if isinstance(top, mb.Compound):
+            top = gmso.external.from_mbuild(top)
+        top.identify_connections()
+
+        if self.backend=='gmso':
+            self._parametrize_gmsoFF(top=top,
+                                typemap=typemap,
+                                combining_rule=combining_rule)
+        else:
+            raise FoyerError('Backend not supported')
+
+        top.update_topology()
+        self._check_parameters(top, assert_bond_params,
+                             assert_angle_params,
+                             assert_dihedral_params,
+                             assert_improper_params,
+                             debug)
+
+        if references_file:
+            atom_types = set(site.atom_type for site in top.sites)
+            self._write_references_to_file(atom_types, references_file)
+        return top
+
+    def _parametrize_gmsoFF(self, top, typemap, combining_rule):
+        """Parametrize a Topology with gmso.ForceField"""
+        # Assign AtomTypes
+        for atom in top.sites:
+            atom.atom_type = deepcopy(self.ff.atom_types.get(
+                                typemap[top.get_index(atom)]['atomtype']))
+        if not all([a.atom_type for a in top.sites]):
```

Comment:
```suggestion
        if not all(a.atom_type for a in top.sites):
```

Reply:
It might be nice to add this function to Topology class itself. Something like 
```py
def is_fully_typed("atoms"):
```
Just a suggestion.